src/atoms_calculators.jl

```diff
@@ -0,0 +1,76 @@
+#
+# Define AtomsCalculators interface for DFTK.
+#
+# This interface is inspired by the one used in Molly.jl, 
+# see https://github.com/JuliaMolSim/Molly.jl/blob/master/src/types.jl
+#
+# TODO: Decide how Hamiltonian terms should be updated upon change of the positions.
+# These terms are held inside the `basis`. Currently, the whole basis is rebuilt.
+# TODO: Find out if symmetries can be re-enabled (see issue on GH).
+
+using AtomsBase
+using AtomsCalculators
+
+Base.@kwdef struct DFTKParameters
+    model_kwargs = (; )
+    basis_kwargs = (; )
+    scf_kwargs = (; )
+end
+
+struct DFTKState
+    scfres::NamedTuple
+end
+function DFTKState(system::AbstractSystem, params::DFTKParameters)
+    basis = prepare_basis(system, params)
+    ρ = guess_density(basis)
```

Comment:
```suggestion
    ρ = guess_density(basis, system)
```
as systems can be magnetic (and this has an influence on the basis)